SECURITY.md

```diff
@@ -0,0 +1,237 @@
+- Each computing unit runs in an isolated Kubernetes pod
+- UI users configure resource limits (CPU, memory, GPU) per pod
+- Pods can be scheduled across multiple nodes for better resource distribution
+
+**Security considerations**:
+- Better isolation between users compared to local computing units
+- Kubernetes provides namespace and pod-level isolation
+- Resource limits prevent individual users from consuming excessive resources
+- Container security and image scanning should be implemented
+- Deployment managers must secure the Kubernetes cluster infrastructure
+
+### What is NOT Guaranteed
+
+Texera's security model does NOT guarantee:
+- Protection against malicious code in user workflows (users can execute arbitrary code)
+- Strong isolation between workflows in local computing units
+- Complete isolation between workflows in Kubernetes computing units within the same namespace
+- Protection against infrastructure-level compromises
+- Protection against deployment manager misconfigurations
+- DDoS protection (requires external infrastructure)
+- Compliance with specific regulatory requirements without additional configuration
+
+## What are NOT Security Issues
+
+The following are **NOT considered security vulnerabilities** in Texera:
+
+### User Code Execution
+
+REGULAR and ADMIN users can execute arbitrary code (Python, R, Scala) within computing units. This is by design - Texera is a data analytics platform where custom code execution is a core feature. The system currently does not sandbox user code beyond the isolation provided by the deployment environment (local processes or Kubernetes pods). Deployment managers should use resource limits, monitor usage, and restrict user roles appropriately.
+
+### Resource Consumption
+
+Users can create workflows that consume significant CPU, memory, or storage. Texera is designed for data-intensive workloads. Deployment managers control this through computing unit resource limits, quotas, and monitoring.
+
+### Information Disclosure within Authorized Access
+
+Users with READ or WRITE access to a resource can view all its contents. Access control is at the resource level - once access is granted, full visibility is expected. Resource owners should grant access only to trusted users.
+
+### Public Resources
+
+Resources marked as public are visible to all users. Public sharing is a deliberate collaboration feature. Users should review resources before making them public and avoid including sensitive data or credentials.
+
+### Issues Requiring Deployment Manager Access
+
+Issues requiring physical access to servers, administrative access to infrastructure, database access, or access to configuration files are out of scope. These access levels are considered trusted.
+
+### Third-Party Dependencies
+
+Theoretical vulnerabilities in dependencies that have not been exploited in Texera's usage are not in scope. Texera regularly updates dependencies and monitors security advisories.
```

Comment:
I'd say that they are welcome to raise an issue and a PR if they wish.

So far, Texera doesn't seem to super active in upgrading dependencies.